### PR TITLE
Automarket UI

### PR DESCRIPTION
**Pull request by dlaxcess**
MANDATORY FOR MERGING :
- [x] add default mint price info in mint modal
- [x] add Price display in Nft details
- [x] add sell button ( Approve & setPrice => refresh after setting price )
- [x] add buy button in Nft details if not owner
- [x] add burn button on Nft detail page ( remove burned nft from store and local storage after burn )
- [x] add Transfer to addresse dEaD if burn function not defined in nft collection
- [x] Disable Transfer button when not effective
- [x] Get Price and royalties infos when not owner to permit buy function working ( with hash of the nft in url on first load nft.collection had only chainId and address. if reloaded brother, nft.collection becomed full hydratated)
- [x] display sell button only on OpenNFTsV4 template
- [x] display "sell on OpenSea" button
- [x] refresh owner and buttons after buy
- [x] display royalty infos on nft mint modale
- [x] add royalty infos fields in collection create and clone collection with thoses params
 
OPTIONAL FEATURES OR AFTER MERGING :
- [ ] Refresh after transfer
- [ ] Burn Nft buyed (owner of nft, not owner of collection)
- [ ] make price component to filter Ether format
- [ ] make royalty percentage component to filter input from % => base point
- [ ] Listen Events after buy to refresh nfts of ex owner of a buyed nft
- [ ] integrate new models from Alex
- [ ] test OpenSea Approve / OpenNFTsV4 approve

### CODE REVIEW

File: common/lib/resolver/resolver-conv-nft-infos.ts [hunk 1]
```diff
@@ -7,13 +7,13 @@ import { IERCNftInfos, IOpenNFTsInfos } from "@soltypes/contracts/interfaces/IOp
 
 const resolverConvNftInfos = (
   chainId: number,
-  collection: CollectionType,
+  collectionOrigin: CollectionType,
```
Comment (zapaz) [inline common/lib/resolver/resolver-conv-nft-infos.ts:10]: why rename ?

File: common/lib/resolver/resolver-conv-nft-infos.ts [hunk 2]
```diff
@@ -24,9 +24,7 @@ const resolverConvNftInfos = (
   const royaltyReceiver = openNFTsInfos[1] || constants.AddressZero;
   const royaltyAmount = openNFTsInfos[2].toString() || "0";
 
-  const collectionName = collection.name || "";
-  const collectionSymbol = collection.symbol || "";
-  const burnable = collection.supports?.IOpenNFTsV4;
+  const collection = collectionOrigin;
```
Comment (zapaz) [inline common/lib/resolver/resolver-conv-nft-infos.ts:27]: ?

File: svelte/components/Nft/Nft.svelte [hunk 3]
```diff
@@ -117,53 +118,55 @@
...
-            <li>
-              <div class="flex"><span class="label">Nft Royalties Amount</span></div>
-              <div class="flex">
-                {#if $nft.royaltyAmount === "0"}
-                  <span class="overflow-ellipsis" title={$nft.royaltyAmount}
-                    >{$nft.price === "0.0" ? "Set price to calculate royalties" : "No royalties amount setted"}</span
-                  >
-                {:else}
+          {#if $nft.collection?.supports?.IOpenMarketable}
+            {#if $nft.price || $nft.price === "0"}
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:122]: better use `parseInt($nft.price) >= 0`

File: svelte/components/Nft/Nft.svelte [hunk 4]
```diff
@@ -50,6 +48,9 @@
 
     // STATE VIEW : sync get Nft
     nft = nftStore.getOneStore(chainId, address, tokenID);
+    if (!$nft?.collection?.supports) {
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:51]: first ? not necassary => `$nft.collection?.supports`

File: svelte/components/Nft/Nft.svelte [hunk 5]
```diff
@@ -181,13 +184,13 @@
               aria-disabled={$nft.price && $nft.price !== "0"}><i class="fa fa-gift" /> Transfer</a
             >
           {/if}
-          {#if $nft.owner === account}
+          {#if $nft.owner === account && $nft.collection?.supports?.IOpenMarketable}
```
Comment (zapaz) [inline svelte/components/Nft/Nft.svelte:187]: first ? to remove , and2 lines further also

**Review by zapaz — CHANGES_REQUESTED** [review]
some small remarks

File: svelte/components/Nft/Nft.svelte (hunk 4 above)
Comment (dlaxcess) [reply svelte/components/Nft/Nft.svelte:50]: Without it it causes an error when using hash of the nft in the url

**Pull request merged**